src/main/java/org/jvnet/hudson/reactor/ReactorListener.java

```diff
@@ -86,23 +87,41 @@ public Aggregator(Collection<ReactorListener> listeners) {
         }
 
         public void onTaskStarted(Task t) {
-            for (ReactorListener listener : listeners)
-                listener.onTaskStarted(t);
+            run(l -> l.onTaskStarted(t));
         }
 
         public void onTaskCompleted(Task t) {
-            for (ReactorListener listener : listeners)
-                listener.onTaskCompleted(t);
+            run(l -> l.onTaskCompleted(t));
         }
 
         public void onTaskFailed(Task t, Throwable err, boolean fatal) {
-            for (ReactorListener listener : listeners)
-                listener.onTaskFailed(t,err,fatal);
+            run(l -> l.onTaskFailed(t,err,fatal));
         }
 
         public void onAttained(Milestone milestone) {
-            for (ReactorListener listener : listeners)
-                listener.onAttained(milestone);
+            run(l -> l.onAttained(milestone));
+        }
+
+        private void run(ListenerAction action) {
+            Error ex = null;
+            for (ReactorListener listener : listeners) {
+                try {
+                    action.run(listener);
+                } catch (Error x) {
```

Comment:
Why only `Error`? RuntimeExceptions won't be caught in such case. Catching `Error`s is probably not required, but runtime exceptions must be caught

Reply:
completely right.